app.js

```diff
@@ -70,17 +72,30 @@ app.post('/sessions', async function(req, res, next) {
     if (process.env['LOG_SINK_URL'])
       request.post({ url: process.env['LOG_SINK_URL'], body: tokenSet, json: true });
 
-    const { groupUri, groupId } = await selectBestuurseenheidByNumber(claims);
+    let { groupUri, groupId } = await selectGroupByNumber(claims);
+
+    const isEconomischeActor = claims.vo_doelgroepcode == "EA"
 
     if (!groupUri || !groupId) {
-      console.log(`User is not allowed to login. No bestuurseenheid found for roles ${JSON.stringify(claims[roleClaim])}`);
-      saveLog(
-        logsGraph,
-        `http://data.lblod.info/class-names/no-bestuurseenheid-for-role`,
-        `User is not allowed to login. No bestuurseenheid found for roles ${JSON.stringify(claims[roleClaim])}`,
-        sessionUri,
-        claims[groupIdClaim]);
-      return res.header('mu-auth-allowed-groups', 'CLEAR').status(403).end();
+      if(isEconomischeActor) {
+        await createEconomischeActorByClaims(claims);
+        ({ groupUri, groupId } = await selectGroupByNumber(claims));  
```

Comment:
Have a look at how this logic (create if it doesn't exist) is implemented in `ensureUserAndAccount` and try to do it in a similar way. Using similar naming will also improve the readability/understandability of the service.